openghg/plotting/_timeseries.py

```diff
@@ -1,61 +1,191 @@
+                    }
+
+    html_string = latex_string
+    for rep in replacements:
+        html_string = html_string.replace(rep,
+                                        replacements[rep])
+
+    return html_string
+
+
+def _plot_remove_gaps(
+        x_data: np.ndarray,
+        y_data: np.ndarray,
+        gap: Optional[int] = 24 * 60 * 60 * 1000000000
+) -> Tuple[np.ndarray, np.ndarray]:
+    '''Insert NaNs between big gaps in the data. 
+    Prevents connecting lines being drawn
+
+    Args:
+        x_data: plot timeseries (numpy timestamp)
+        y_data: data array
+        gap: gap beyond which a NaN is introducted (nanoseconds, default=1 day)
+
+    Returns:
+        x, y: x and y arrays to plot
+    '''
+
+    gap_idx = np.where(np.diff(x_data.astype(int)) > gap)[0]
+    x_data_plot = np.insert(x_data, gap_idx + 1, values=x_data[0])
+    y_data_plot = np.insert(y_data, gap_idx + 1, values=np.nan)
+
+    return x_data_plot, y_data_plot
+
+
+def _plot_legend_position(
+        ascending: bool
+) -> Tuple[Dict, Dict]:
+    """Position of legend and logo,
+    depending on whether data is ascending or descending
+
+    Args:
+        ascending (bool): Is the data ascending
+
+    Returns:
+        Dict, Dict: Plotly legend and logo position parameters
+    """
+
+    if ascending:
+        legend_pos = {
+            "yanchor": "top",
+            "xanchor": "left",
+            "y": 0.99,
+            "x": 0.01
+        }
+        logo_pos = {
+            "yanchor": "bottom",
+            "xanchor": "right",
+            "y": 0.01,
+            "x": 0.99
+        }
+    else:
+        legend_pos = {
+            "yanchor": "top",
+            "xanchor": "right",
+            "y": 0.99,
+            "x": 0.99
+        }
+        logo_pos = {
+            "yanchor": "bottom",
+            "xanchor": "left",
+            "y": 0.01,
+            "x": 0.01
+        }
+
+    return legend_pos, logo_pos
+
+
+def _plot_logo(
+        logo_pos: Dict,    
+) -> Dict:
+    """Create Plotly dictionary for logo
+
+    Args:
+        logo_pos: Dictionary containing the position of the logo
+
+    Returns:
+        Dictionary containing logo + position parameters
+    """
+
+    logo = base64.b64encode(open(get_datapath("OpenGHG_Logo_NoText_transparent_200x200.png"), 'rb').read())
+    logo_dict = dict(
+        source='data:image/png;base64,{}'.format(logo.decode()),
+        xref="x domain",
+        yref="y domain",
+        sizex=0.1,
+        sizey=0.1)
+    logo_dict.update(logo_pos)
+
+    return logo_dict
 
 
 def plot_timeseries(
     data: Union[ObsData, List[ObsData]],
-    title: Optional[str] = None,
     xvar: Optional[str] = None,
     yvar: Optional[str] = None,
     xlabel: Optional[str] = None,
     ylabel: Optional[str] = None,
     units: Optional[str] = None,
+    logo: Optional[bool] = True,
 ) -> go.Figure:
     """Plot a timeseries
 
     Args:
         data: ObsData object or list of objects
```

Comment:
Was there a reason to remove the title from the figure? This does mean the figure can be made more explicit about what is being displayed?

Reply:
No. I think I forgot to add it back in. Happy to put an optional title back.

Reply:
Yeah I think having an option to add a title is good.